src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -130,19 +136,24 @@ export default class PhotoField extends React.Component {
               this.props.onChange({
                 errorMessage: 'The photo you uploaded isn’t a square photo. Please upload a new one that fits the requirements.'
               });
+              this.setState({ currentLayout: layouts.screenReaderError });
             } else if (!isValidImageSize(img)) {
               this.props.onChange({
                 errorMessage: 'The file you selected is smaller than the 350-pixel minimum file width or height and couldn’t be uploaded. Please try to upload a different photo.'
               });
+
+              this.setState({ currentLayout: layouts.screenReaderError });
             } else {
-              this.setState({ progress: 0, warningMessage: null });
+              this.setState({
+                currentLayout: layouts.watchUpload,
+                progress: 0,
+                warningMessage: null
+              });
               this.uploadRequest = this.props.formContext.uploadFile(
                 file,
                 this.props.uiSchema['ui:options'],
                 this.updateProgress,
-                (fileData) => {
-                  this.props.onChange(fileData);
-                },
+                (formData) => this.handleUploadComplete(formData, file),
                 () => {
                   this.props.onBlur(this.props.idSchema.$id);
                   this.uploadRequest = null;
```

Comment:
What is this callback for? It's on both of the `onChange` handlers- I'm not sure how to update the layout state for this scenario.

Reply:
This for when an error happens. I don't think you will need to do anything in here.